wunderlist-create-task.js

```diff
@@ -9,19 +9,49 @@ module.exports = function(RED) {
     this.on('input', function(msg) {
       var wunderlistAPI = wunderlistSDK.getApi(node.config, msg);
       var tasks = wunderlistAPI.http.tasks;
+      var reminders = wunderlistAPI.http.reminders;
 
-      tasks.create({
+      var params = {
         'list_id': Number(n.listId || msg.listId),
         'title': n.taskTitle || msg.payload
-      })
+      };
+      // Add due date if in msg
+      if (msg.due_date){
+        params.due_date = msg.due_date;
+      }
+
+      tasks.create(params)
       .done(function (taskData, statusCode) {
         msg.taskId = taskData.id;
-        node.send(msg);
+        msg.task = taskData;
+
+        // If a reminder is defined
+        if (msg.reminder_date){
+          var reminderData = {
+            'task_id' : taskData.id,
+            'date' : msg.reminder_date
+          }
+          reminders.create(reminderData)
+          .done((reminderData, status) => {
+            msg.reminder = reminderData;
+            node.send(msg);
+          })
+          .fail((resp,code) => {
+            //Delete if it fails to create reminder
+            tasks.deleteID(taskData.id, 1)
```

Comment:
I don't see a reason to delete the task if setting the reminder fails.
an error is reported to the user so he\she should decide how to handle it.

Reply:
Ok, adjusted